src/analysis/souffle/test_taint_with_auth_logic.dl

```diff
@@ -0,0 +1,99 @@
+//-----------------------------------------------------------------------------
```

Comment:
Could you make a tests subdirectory for authorization logic tests? (perhaps `tests/authorization_logic` or similar)?